File: app/src/main/java/com/hfad/avc/ui/template/СhangeTemplateFragment.java
package com.hfad.avc.ui.template;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import androidx.appcompat.widget.Toolbar;
import androidx.databinding.DataBindingUtil;

import com.google.android.material.snackbar.BaseTransientBottomBar;
import com.google.android.material.snackbar.Snackbar;
import com.hfad.avc.R;
import com.hfad.avc.data.model.Template;
import com.hfad.avc.databinding.FragmentTemplateBinding;

import moxy.MvpAppCompatFragment;
import moxy.presenter.InjectPresenter;
import moxy.presenter.ProvidePresenter;


public class СhangeTemplateFragment extends MvpAppCompatFragment implements IСhangeTemplateViewModel {

   @InjectPresenter
   СhangeTemplatePresenter presenter;

   private Toolbar toolbar;
   private MenuItem favoriteTemplate;
   private MenuItem changeTemplate;
   private EditText textTemplate;
   private FragmentTemplateBinding binding;

   @ProvidePresenter
   СhangeTemplatePresenter ProvidePresenterContactPresenter() {
      return new СhangeTemplatePresenter(getArguments());
   }

   @Override
   public View onCreateView(LayoutInflater inflater, ViewGroup container,
                            Bundle savedInstanceState) {
      binding = DataBindingUtil.inflate(inflater, R.layout.fragment_template, container, false);
      final View view = binding.getRoot();
      this.toolbar = view.findViewById(R.id.toolbar);
      this.toolbar.inflateMenu(R.menu.menu_contact);
      this.toolbar.setNavigationOnClickListener(v -> requireActivity().onBackPressed());
      this.favoriteTemplate = toolbar.getMenu().findItem(R.id.favoriteObject);
      this.changeTemplate = toolbar.getMenu().findItem(R.id.changeObject);
      this.textTemplate = binding.textTemplate;
      this.presenter.setFavoriteTemplate(favoriteTemplate);
      this.favoriteTemplate.setOnMenuItemClickListener(v -> {
         this.presenter.setOnClickFavoriteTemplate();
         this.presenter.setFavoriteTemplate(favoriteTemplate);
         this.presenter.updateTemplateDB();
         return false;
      });
      this.changeTemplate.setOnMenuItemClickListener(v -> {
         boolean isCursorVisible =this.textTemplate.isCursorVisible();
         if (isCursorVisible) {
            this.presenter.updateTemplateDB();
         }
         this.textTemplate.setCursorVisible(!isCursorVisible);
         this.textTemplate.setFocusable(!isCursorVisible);
         this.textTemplate.setClickable(!isCursorVisible);

         return false;
      });
      return view;
   }

   @Override
   public void onCreate(Bundle savedInstanceState) {
      super.onCreate(savedInstanceState);
      setHasOptionsMenu(true);
   }

   @Override
   public void setData(Template template) {
      this.binding.setTemplateDetail(template);
      this.binding.setPresenterTemplate(this.presenter);
   }

   @Override
   public void onInsertDB(String newId) {
      this.presenter.setNewTemplateText();
   }

   @Override
   public void sendOkUsers() {
      Snackbar.make(requireView(), "Сохранено", BaseTransientBottomBar.LENGTH_LONG).show();
   }
}

File: app/src/main/java/com/hfad/avc/ui/template/СhangeTemplateFragment.kt
package com.hfad.avc.ui.template

import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import androidx.appcompat.widget.Toolbar
import androidx.databinding.DataBindingUtil
import com.google.android.material.snackbar.BaseTransientBottomBar
import com.google.android.material.snackbar.Snackbar
import com.hfad.avc.R
import com.hfad.avc.data.model.Template
import com.hfad.avc.databinding.FragmentTemplateBinding
import moxy.MvpAppCompatFragment
import moxy.presenter.InjectPresenter
import moxy.presenter.ProvidePresenter


class СhangeTemplateFragment : MvpAppCompatFragment(), IСhangeTemplateViewModel {
   @InjectPresenter
   lateinit var presenter: СhangeTemplatePresenter
   lateinit var toolbar: Toolbar
   lateinit var favoriteTemplate: MenuItem
   lateinit var changeTemplate: MenuItem
   lateinit var textTemplate: EditText
   lateinit var binding: FragmentTemplateBinding

   @ProvidePresenter
   fun ProvidePresenterContactPresenter(): СhangeTemplatePresenter {
      return СhangeTemplatePresenter(arguments)
   }

   override fun onCreateView(
      inflater: LayoutInflater, container: ViewGroup?,
      savedInstanceState: Bundle?,
   ): View {
      this.binding = DataBindingUtil.inflate(inflater, R.layout.fragment_template, container, false)
      val view: View = binding.root
      this.toolbar = view.findViewById(R.id.toolbar)
      this.toolbar.inflateMenu(R.menu.menu_contact)
      this.toolbar.setNavigationOnClickListener { requireActivity().onBackPressed() }
      this.favoriteTemplate = toolbar.menu.findItem(R.id.favoriteObject)
      this.changeTemplate = toolbar.menu.findItem(R.id.changeObject)
      this.textTemplate = binding.textTemplate
      this.favoriteTemplate.let { presenter.setFavoriteTemplate(it) }
      this.favoriteTemplate.setOnMenuItemClickListener {
         presenter.setOnClickFavoriteTemplate()
         presenter.setFavoriteTemplate(favoriteTemplate)
         presenter.updateTemplateDB()
         false
      }
      this.changeTemplate.setOnMenuItemClickListener {
         val isCursorVisible = textTemplate.isCursorVisible
         if (isCursorVisible) {
            presenter.updateTemplateDB()
         }
         textTemplate.isCursorVisible = !isCursorVisible
         textTemplate.isFocusable = !isCursorVisible
         textTemplate.isClickable = !isCursorVisible
         false
      }
      return view
   }

   override fun onCreate(savedInstanceState: Bundle?) {
      super.onCreate(savedInstanceState)
      setHasOptionsMenu(true)
   }

   override fun setData(template: Template?) {
      this.binding.templateDetail = template
      this.binding.presenterTemplate = presenter
   }

   override fun onInsertDB(newId: String?) {
      this.presenter.setNewTemplateText()
   }

   override fun sendOkUsers() {
      Snackbar.make(requireView(), "Сохранено", BaseTransientBottomBar.LENGTH_LONG).show()
   }
}